### PR TITLE
[fix] share account capitalization and share_grant accounts spurious diffs

### PR DESCRIPTION
**Pull request by alldoami**

### CODE REVIEW

**Comment by alldoami** [comment]
https://github.com/hashicorp/terraform-plugin-sdk/issues/160

**Comment by alldoami** [comment]
To fix check-docs build failure, run `make docs`

File: pkg/resources/share.go [hunk 1]
```diff
@@ -28,10 +28,11 @@ var shareSchema = map[string]*schema.Schema{
 		Description: "Specifies a comment for the managed account.",
 	},
 	"accounts": &schema.Schema{
-		Type:        schema.TypeSet,
-		Elem:        &schema.Schema{Type: schema.TypeString},
-		Optional:    true,
-		Description: "A list of accounts to be added to the share.",
+		Type:             schema.TypeList,
```
Comment (edulop91) [inline pkg/resources/share.go]: can you add a comment linking that github issue you found last night?

Comment (alldoami) [reply pkg/resources/share.go]: ah ok I thought you meant in the PR

Comment (edulop91) [reply pkg/resources/share.go]: I feel like this makes it a breaking change, can you prefix your pull request title with `[breaking]`

File: pkg/resources/share.go [hunk 2]
```diff
@@ -28,10 +28,12 @@ var shareSchema = map[string]*schema.Schema{
 		Description: "Specifies a comment for the managed account.",
 	},
 	"accounts": &schema.Schema{
-		Type:        schema.TypeSet,
-		Elem:        &schema.Schema{Type: schema.TypeString},
-		Optional:    true,
-		Description: "A list of accounts to be added to the share.",
+		// Changed from Set to List to use DiffSuppressFunc: hashicorp/terraform-plugin-sdk#160
```
Comment (edulop91) [inline pkg/resources/share.go]:
```suggestion
		// Changed from Set to List to use DiffSuppressFunc: https://github.com/hashicorp/terraform-plugin-sdk/issues/160
```

File: pkg/resources/grant_helpers.go [hunk 3]
```diff
@@ -131,7 +131,7 @@ func readGenericGrant(data *schema.ResourceData, meta interface{}, builder snowf
 				continue
 			}
 
-			shares = append(shares, grant.GranteeName)
+			shares = append(shares, granteeNameStrippedAccount)
```
Comment (edulop91) [inline pkg/resources/grant_helpers.go]: i wonder if there's an easy way to create a test for this

**Pull request merged**